chain/jsonrpc/src/api/transactions.rs

```diff
@@ -3,28 +3,52 @@ use serde_json::Value;
 use near_client_primitives::types::TxStatusError;
 use near_jsonrpc_primitives::errors::RpcParseError;
 use near_jsonrpc_primitives::types::transactions::{
-    RpcBroadcastTransactionRequest, RpcTransactionError, RpcTransactionStatusCommonRequest,
-    TransactionInfo,
+    RpcSendTransactionRequest, RpcTransactionError, RpcTransactionStatusRequest, TransactionInfo,
 };
 use near_primitives::borsh::BorshDeserialize;
 use near_primitives::transaction::SignedTransaction;
 
 use super::{Params, RpcFrom, RpcRequest};
 
-impl RpcRequest for RpcBroadcastTransactionRequest {
+impl RpcRequest for RpcSendTransactionRequest {
     fn parse(value: Value) -> Result<Self, RpcParseError> {
-        let signed_transaction =
-            Params::new(value).try_singleton(|value| decode_signed_transaction(value)).unwrap()?;
-        Ok(Self { signed_transaction })
+        println!("beginning of parsing RpcSendTransactionRequest");
+        let tx_request = Params::new(value)
+            .try_singleton(|value| {
+                println!("parsing RpcSendTransactionRequest - try singleton");
+                Ok(RpcSendTransactionRequest {
+                    signed_transaction: decode_signed_transaction(value)?,
+                    // will be ignored in `broadcast_tx_async`, `broadcast_tx_commit`
+                    wait_until: Default::default(),
+                })
+            })
+            .try_pair(|_: String, _: String| {
+                // Here, we restrict serde parsing object from the array
```

Comment:
I'm always surprised that serde can parse objects from the arrays by default.
I restrict here the option to use new features from legacy format.